sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/FullFidelityChangeFeedProcessorBuilder.java

```diff
@@ -0,0 +1,179 @@
+public class FullFidelityChangeFeedProcessorBuilder {
+    private String hostName;
+    private CosmosAsyncContainer feedContainer;
+    private CosmosAsyncContainer leaseContainer;
+    private ChangeFeedProcessorOptions changeFeedProcessorOptions;
+    private Consumer<List<ChangeFeedProcessorItem>> consumer;
+
+    /**
+     * Instantiates a new Cosmos a new ChangeFeedProcessor builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder() {
+    }
+
+    /**
+     * Sets the host name.
+     *
+     * @param hostName the name to be used for the host. When using multiple hosts, each host must have a unique name.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder hostName(String hostName) {
+        this.hostName = hostName;
+        return this;
+    }
+
+    /**
+     * Sets and existing {@link CosmosAsyncContainer} to be used to read from the monitored container.
+     *
+     * @param feedContainer the instance of {@link CosmosAsyncContainer} to be used.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder feedContainer(CosmosAsyncContainer feedContainer) {
+        this.feedContainer = feedContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets an existing {@link CosmosAsyncContainer} to be used to read from the leases container.
+     *
+     * @param leaseContainer the instance of {@link CosmosAsyncContainer} to use.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder leaseContainer(CosmosAsyncContainer leaseContainer) {
+        this.leaseContainer = leaseContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets a consumer function which will be called to process changes.
+     *
+     * <!-- src_embed com.azure.cosmos.fullFidelityChangeFeedProcessor.handleChanges -->
+     * <pre>
+     * .handleChanges&#40;docs -&gt; &#123;
+     *     for &#40;JsonNode item : docs&#41; &#123;
+     *         &#47;&#47; Implementation for handling and processing of each JsonNode item goes here
+     *     &#125;
+     * &#125;&#41;
+     * </pre>
+     * <!-- end com.azure.cosmos.fullFidelityChangeFeedProcessor.handleChanges -->
+     *
+     * @param consumer the {@link Consumer} to call for handling the feeds.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder handleChanges(Consumer<List<ChangeFeedProcessorItem>> consumer) {
```

Comment:
Similar to what Annie mentioned - should be generic? To allow passing in types as well for payloads at least?

Reply:
going ahead with `JsonNode` instead of `<T>` - will update the documentation.